website/docs/r/upcloud_server.html.markdown

```diff
@@ -56,34 +52,22 @@ The following arguments are supported:
 
 * `hostname` - (Required) A valid domain name, e.g. host.example.com. The maximum length is 128 characters.
 * `zone` - (Required) - The zone in which the server will be hosted, e.g. fi-hel1. See [Zones API](https://developers.upcloud.com/1.3/5-zones/)
+* `network_interface` - (Required) One or more blocks describing the network interfaces of the server. Any changes to these blocks will force the creation of a new server resource.
 * `firewall` - (Optional) Are firewall rules active for the server
 * `metadata` - (Optional) Is the metadata service active for the server
 * `cpu` - (Optional) The number of CPU for the server
 * `mem` - (Optional) The size of memory for the server
 * `template` - (Optional) The template to use during creation
 * `user_data` - (Optional) Defines URL for a server setup script, or the script body itself
 * `plan` - (Optional) The pricing plan used for the server
-* `storage_devices` - (Required) A list of storage devices associated with the server
+* `storage_devices` - (Optional) A list of storage devices associated with the server
 * `login` - (Optional) Configure access credentials to the server
-* `network_interface` - (Required) One or more blocks describing the network interfaces of the server. Any changes to these blocks will force the creation of a new server resource.
 
 The `storage_devices` block supports:
```

Comment:
seems that the template block is missing similar description

Reply:
Good catch! Added now ☺️